game/scripts/vscripts/abilities/oaa_zuus_cloud.lua

```diff
@@ -0,0 +1,114 @@
+zuus_cloud = class( AbilityBaseClass )
+LinkLuaModifier( "modifier_zuus_cloud_oaa", "abilities/oaa_zuus_cloud.lua", LUA_MODIFIER_MOTION_NONE )
+
+function zuus_cloud:OnSpellStart()
+  local caster = self:GetCaster()
+  local hCloud = CreateUnitByName( "npc_dota_zeus_cloud", self:GetCursorPosition(), true, caster, caster, caster:GetTeamNumber() )
+  hCloud:SetOwner( self:GetCaster() )
+  hCloud:SetControllableByPlayer( self:GetCaster():GetPlayerOwnerID(), false )
+  hCloud:AddNewModifier( caster, self, "modifier_zuus_cloud_oaa", { Interval = self:GetSpecialValueFor( "cloud_bolt_interval" ), Radius = self:GetSpecialValueFor( "cloud_radius" ) } )
+  hCloud:AddNewModifier( caster, self, "modifier_kill", { duration = self:GetSpecialValueFor( "cloud_duration" ) } )
+  FindClearSpaceForUnit( hCloud, self:GetCursorPosition(), true )
```

Comment:
I don't think this is really necessary since the Nimbus should be flying and have no collision. On that note though, you should check if it has proper behaviour when placed over trees and that you can walk under it.

Reply:
FindClearSpaceForUnit does not change unit colision, it is just to place it.
it has no effect.
It does work on trees and you can walk pass it

Reply:
It has proper behavior. I can place it over trees and the trees don't get destroyed and i can walk underneath the nimbus

Reply:
I know it doesn't affect collision, I just meant there isn't really any point to doing this when it shouldn't need "clear space" since it has no collision. And now I realised that actually this should already be taken of by `CreateUnitByName` in the first place.